src/tdastro/effects/redshift.py

```diff
@@ -0,0 +1,76 @@
+from tdastro.base_models import EffectModel
+
+
+class Redshift(EffectModel):
+    """A redshift effect model.
+
+    This contains a "pre-effect" method, which is used to calculate the emitted wavelengths/times
+    needed to give us the observed wavelengths and times given the redshift.
+
+    Attributes
+    ----------
+    pz : `float`
+        The redshift.
+
+    Notes
+    -----
+    Conversions used are as follows:
+    - emitted_wavelength = observed_wavelength / (1 + redshift)
+    - emitted_time = observed_time / (1 + redshift)
+    - observed_flux = emitted_flux / (1 + redshift)
+    """
+
+    def __init__(self, pz, **kwargs):
+        """Create a Redshift effect model.
+
+        Parameters
+        ----------
+        pz : `float`
```

Comment:
Minor comment: I think `pz` means photometric `z`, it would be better to use `z` or `redshift` here. I personally prefer longer variable names, so I vote for `redshift`